testing/openjdk12/APKBUILD

```diff
@@ -0,0 +1,307 @@
+subpackages="$pkgname-jmods:_jmods:noarch
+$pkgname-demos:_demos:noarch
+$pkgname-doc:_doc:noarch
+$pkgname-dbg:_dbg
+$pkgname-jre:_jre
+$pkgname-src:_src:noarch
+$pkgname-jre-headless:_jre_headless
+$pkgname-jdk:_jdk"
+source="jdk-$_pkgver.tar.bz2::http://hg.openjdk.java.net/jdk-updates/jdk12u/archive/jdk-$_pkgver.tar.bz2
+
+build.patch
+aarch64.patch
+arm.patch
+ppc64le.patch
+x86.patch
+
+HelloWorld.java
+TestECDSA.java
+TestCryptoLevel.java
+Alpine_Bug_10126.java
+"
+builddir="$srcdir/jdk12u-jdk-$_pkgver"
+
+_java_home="/usr/lib/jvm/java-12-openjdk"
+
+ldpath="$_java_home/lib:$_java_home/lib/server"
+sonameprefix="$pkgname:"
+
+# enable running the JTReg tests in check?
+# see comment in that function for explanation
+_run_jtreg=${_run_jtreg:-0}
+if [ $_run_jtreg -ne 0 ]; then
+	makedepends="$makedepends java-jtreg"
+	checkdepends="$checkdepends ttf-freefont xvfb-run"
+fi
+
+
+prepare() {
+	default_prepare
+
+	# update autoconf files to detect alpine
+	update_config_sub
+	update_config_guess
+
+	# remove not compilable module (hotspot jdk.hotspot.agent)
+	# this needs libthread_db which is only provided by glibc
+	#
+	# haven't found any way to disable this module so just remove it.
+	rm -r src/jdk.hotspot.agent
+}
+
+build() {
+	if [ $_run_jtreg -ne 0 ]; then
+		_with_jtreg="--with-jtreg=/usr/share/java/jtreg"
+	else
+		_with_jtreg="--with-jtreg=no"
+	fi
+
+	# CFLAGS, CXXFLAGS and LDFLAGS are ignored as shown by a warning
+	# in the output of ./configure unless used like such:
+	#  --with-extra-cflags="${CFLAGS}"
+	#  --with-extra-cxxflags="${CXXFLAGS}"
+	#  --with-extra-ldflags="${LDFLAGS}"
+	# See also paragraph "Configure Control Variables" from "common/doc/building.md"
+	# shellcheck disable=2097 disable=2098
+	CFLAGS='' CXXFLAGS='' LDFLAGS='' \
+		bash ./configure \
+		--build=$CBUILD \
+		--host=$CHOST \
+		--target=$CTARGET \
+		--prefix="$_java_home" \
+		--sysconfdir=/etc \
+		--mandir=/usr/share/man \
+		--infodir=/usr/share/info \
+		--localstatedir=/var \
+		--with-extra-cflags="$CFLAGS" \
+		--with-extra-cxxflags="$CXXFLAGS" \
+		--with-extra-ldflags="$LDFLAGS" \
+		--with-zlib=system \
+		--with-libjpeg=system \
+		--with-giflib=system \
+		--with-libpng=system \
+		--with-lcms=system \
+		--with-jobs=${JOBS:-4} \
+		--with-test-jobs=${JOBS:-4} \
+		--with-native-debug-symbols=external \
+		$_with_jtreg \
+		--disable-warnings-as-errors \
+		--disable-precompiled-headers \
+		--enable-dtrace=no \
+		--with-jvm-variants=server \
+		--with-debug-level=release \
+		--with-version-pre= \
+		--with-version-opt=alpine-r${pkgrel} \
+		--with-version-build=${_pkgver#*+} \
+		--with-vendor-name="Alpine" \
+		--with-vendor-url="https://alpinelinux.org/" \
+		--with-vendor-bug-url="https://bugs.alpinelinux.org/projects/alpine/issues" \
+		--with-vendor-vm-bug-url="https://bugs.alpinelinux.org/projects/alpine/issues"
+	MAKEFLAGS='' make jdk-image
```

Comment:
Why assign `MAKEFLAGS` here?

Reply:
When `MAKEFLAGS` are specified, the build fails as it cannot handle the `-j` option. So setting it to empty makes the build work

Reply:
It sounds like adding `-j 1` may be a better solution.